service/src/main/java/bio/terra/workspace/service/resource/AwsResourceValidationUtils.java

```diff
@@ -10,6 +10,7 @@
 public class AwsResourceValidationUtils {
   static final Pattern s3ObjectDisallowedChars = Pattern.compile("[{}^%`<>~#|@*+\\[\\]\"\'\\\\/]");
 
+  // TODO(TERRA-523) Is this validation needed for resource name?
```

Comment:
I think this will still be useful to fail fast on an invalid folder path.
For the WSM resource name, use `ResourceValidationUtils.validateResourceName()`

Reply:
Noted. 
Reason for this comment / todo: is this is used for body.common.name, we may relax naming checks since it wont be used for the cloud. 
cloud name will likely be in body.awsStorageFolder.folderName, which will anyway need to validated with a similar function